### DISCUSSION TITLE
DataGrid css isolation RowClass

### DISCUSSION THREAD

**Discussion by avojacek** [microsoft/fluentui-blazor#1757] [Q&A] [answered]
Hi is there any way to use css isolation with RowClass in datagrid?
Im using OnRowFocus in DataGrid to manage current row (selected row) and RowClass to highlight selected row by seting aditional class for currently selected row. 
This works ok while selected-row class is in App.css when I try to move that class in component css (for leverage css isolation) It stops working (that class from isolated css gets random identifier) which I do not know how to use in RowClass code.

`
private IbexCaseDb selectedRow;
	
private async Task RowFocus(FluentDataGridRow<IbexCaseDb> row)
{
if (selectedRow == row.Item)
	return;

selectedRow = row.Item;

await OnSelected.InvokeAsync(row.Item!);
}
private String RowClass(IbexCaseDb arg)
{
	string classes = "";
	classes = arg.analyzed is true ? "highlighted-row" : "";
	if (arg.Id == selectedRow?.Id)
		classes += " selected-row";
	return classes;
}

`
Is there any way to use RowClass with CSS isolation so that classes for DataGrid are isolated from global css code?

Thank you Ales

**Comment by avojacek** [comment]
May be that this is same like #1739

**Comment by avojacek** [comment, marked as answer]
So if I need to have CSS isolation for my component and use classes from isolated CSS inside component code. I have to se custom css scope like this
```
<ItemGroup>
  <None Update="Components/Pages/*.razor.css" CssScope="custom-scope-identifier" />
</ItemGroup>
```

**Reply by Lindsay-Mathieson** [reply]
> So if I need to have CSS isolation for my component and use classes from isolated CSS inside component code. I have to se custom css scope like this
> 
> ```
> <ItemGroup>
>   <None Update="Components/Pages/*.razor.css" CssScope="custom-scope-identifier" />
> </ItemGroup>
> ```

I don't understand how this works? Is there mroe required to support a custom scope?

I just put the above in my project and it broke the working isolation css files and didn't fix the one I had for a DataGrid row class
